ludwig/data/dataset/ray.py

```diff
@@ -174,14 +137,46 @@ def data_format(self):
 class RayDatasetShard(Dataset):
     def __init__(
         self,
-        dataset_shard: DatasetPipeline,
+        dataset_shard: _Dataset,
         features: Dict[str, Dict],
         training_set_metadata: Dict[str, Any],
+        data_loader_kwargs: Dict[str, Any],
     ):
         self.dataset_shard = dataset_shard
         self.features = features
         self.training_set_metadata = training_set_metadata
-        self.epoch_iter = dataset_shard.iter_epochs()
+        self.epoch_iter = self.pipeline(**data_loader_kwargs)
+
+    def pipeline(
+        self,
+        shuffle: bool = True,
+        fully_executed: bool = True,
+        window_size_bytes: Optional[int] = None,
+        shuffle_seed: int = default_random_seed,
+    ) -> DatasetPipeline:
```

Comment:
Had to move this into RayDatasetShard so that we can still keep granular control over how we want too set the DatasetPipeline objects for train/test/val. The alternative was to do this from the DatasetConfig, but that would require always enabling windowing, which likely will cause performance degradation in terms of model quality.

My suspicion for the increase in 2 CPUs is from moving this code block from RayDataset to RayDatasetShard, since we're now working on the split dataset, but need to validate and understand why.